src/v/cluster/partition_balancer_planner.cc

```diff
@@ -890,10 +917,10 @@ void partition_balancer_planner::get_rack_constraint_repair_actions(
 
     for (const auto& ntp : ctx.state().ntps_with_broken_rack_constraint()) {
```

Comment:
would be nice to have yields in this loop too (although we should be careful because this collection can change with a topic table update)

Reply:
we co_await in L915, why yield again?

Reply:
Will it actually yield if `ctx.with_partition` doesn't yield?

Reply:
> Will it actually yield if ctx.with_partition doesn't yield?

looks like it will - even `co_await ss::now()` yields!